Sources/Sentry/SentrySdkInfo.m

```diff
@@ -0,0 +1,63 @@
+#import "SentrySdkInfo.h"
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@implementation SentrySdkInfo
+
+- (instancetype)initWithName:(NSString *)name andVersion:(NSString *)version
+{
+    if (self = [super init]) {
+
+        if ([name length] == 0) {
+            _name = @"";
+        } else {
+            _name = name;
+        }
+
+        if ([version length] == 0) {
+            _version = @"";
+        } else {
+            _version = version;
+        }
+    }
+
+    return self;
+}
+
+- (instancetype)initWithDict:(NSDictionary *)dict
+{
+    if (self = [super init]) {
+
+        if (nil == dict) {
+            _name = @"";
+            _version = @"";
+            return self;
+        }
+
+        if (nil != dict[@"sdk"] && [dict[@"sdk"] isKindOfClass:[NSDictionary class]]) {
+            NSDictionary<NSString *, id> *sdkInfoDict = dict[@"sdk"];
+            if ([sdkInfoDict[@"name"] isKindOfClass:[NSString class]]) {
+                _name = sdkInfoDict[@"name"];
+            }
```

Comment:
M: looks like we're not setting `_name` to `@""` now on the `else` block here.
What about we take `name` and `version` as local vars and call the overload init (the one on line 10, its marked as `NS_DESIGNATED_INITIALIZER` in the header file so it should always be called) at the end of the method?
That one will take care of validating each `NSString` already and it's always good to have a single ctor code funnels to